tests/python/test_multidevice.py

```diff
@@ -112,3 +113,362 @@ def multidevice_schedule(self):
+        T131 = self.ops.permute(T130, dims=[0, 2, 1, 3])
+        T137 = self.ops.reshape(T117, new_shape=[b, s, a, h // a])
+        T138 = self.ops.permute(T137, dims=[0, 2, 1, 3])
+        S139 = self.define_scalar(0.100000, dtype=DataType.Double)
+        S140 = self.define_scalar(True, dtype=DataType.Bool)
+        T141, T142, T143, T144 = self.ops.sdpfa_fwd(T131, T124, T138, S139, S140, None)
+        T145 = self.ops.permute(T141, dims=[0, 2, 1, 3])
+        T146 = self.ops.stride_order(T145, stride_order=[3, 2, 1, 0])
+        T151 = self.ops.reshape(T146, new_shape=[b, s, h])
+        T152 = self.ops.linear(T151, self.mha_linear1_weight, self.mha_linear1_bias)
+        T153 = self.ops.cast(T152, dtype=DataType.Float)
+        T154 = self.ops.cast(T33, dtype=DataType.Float)
+        T155 = self.ops.mul(T153, T154)
+        S156 = self.define_scalar(1.11111, dtype=DataType.Double)
+        T157 = self.ops.mul(T155, S156)
+        T158 = self.ops.add(T31, T157)
+        T159, T160 = self.ops.var_mean(T158, dims=[2], correction=0, keepdim=False)
+        T165 = self.ops.broadcast_in_dim(T159, shape=[b, s, 1], broadcast_dims=[0, 1])
+        T170 = self.ops.broadcast_in_dim(T160, shape=[b, s, 1], broadcast_dims=[0, 1])
+        S171 = self.define_scalar(1.00000e-05, dtype=DataType.Double)
+        T172 = self.ops.add(T165, S171)
+        T177 = self.ops.broadcast_in_dim(
+            T170, shape=[b, s, h], broadcast_dims=[0, 1, 2]
+        )
+        T178 = self.ops.rsqrt(T172)
+        T179 = self.ops.sub(T158, T177)
+        T184 = self.ops.broadcast_in_dim(
+            T178, shape=[b, s, h], broadcast_dims=[0, 1, 2]
+        )
+        T185 = self.ops.mul(T179, T184)
+        T190 = self.ops.broadcast_in_dim(
+            self.layernorm1_weight, shape=[b, s, h], broadcast_dims=[2]
+        )
+        T191 = self.ops.cast(T190, dtype=DataType.Float)
+        T192 = self.ops.mul(T185, T191)
+        T197 = self.ops.broadcast_in_dim(
+            self.layernorm1_bias, shape=[b, s, h], broadcast_dims=[2]
+        )
+        T198 = self.ops.cast(T197, dtype=DataType.Float)
+        T199 = self.ops.add(T192, T198)
+        T200 = self.ops.cast(T199, dtype=DataType.BFloat16)
+        T201 = self.ops.linear(T200, self.mlp_linear0_weight, self.mlp_linear0_bias)
+        T202 = self.ops.cast(T201, dtype=DataType.Float)
+        T203 = self.ops.mul(T202, T202)
+        T204 = self.ops.mul(T203, T202)
+        S205 = self.define_scalar(0.0447150, dtype=DataType.Double)
+        T206 = self.ops.mul(S205, T204)
+        T207 = self.ops.add(T202, T206)
+        S208 = self.define_scalar(0.797885, dtype=DataType.Double)
+        T209 = self.ops.mul(S208, T207)
+        T210 = self.ops.tanh(T209)
+        S211 = self.define_scalar(0.500000, dtype=DataType.Double)
+        T212 = self.ops.mul(S211, T202)
+        S213 = self.define_scalar(1.00000, dtype=DataType.Double)
+        T214 = self.ops.add(S213, T210)
+        T215 = self.ops.mul(T212, T214)
```

Comment:
please write the definition of GELU as a separate function

Reply:
ditto